lib/kernel-manager.coffee

```diff
@@ -140,55 +146,59 @@ class KernelManager
             return spec?.language and spec.display_name and spec.argv
 
 
-    saveKernelSpecs: (jsonString) ->
-        console.log 'saveKernelSpecs:', jsonString
-
-        try
-            newKernelSpecs = JSON.parse(jsonString).kernelspecs
-
-        catch e
-            message =
-                'Cannot parse `ipython kernelspecs` or `jupyter kernelspecs`'
-            options = detail:
-                'Use kernelSpec option in Hydrogen or update IPython/Jupyter to
-                a version that supports: `jupyter kernelspec list --json` or
-                `ipython kernelspec list --json`'
-            atom.notifications.addError message, options
-            return
-
-        unless newKernelSpecs?
-            return
-
-        kernelSpecs = @parseKernelSpecSettings()
-        _.assign kernelSpecs, newKernelSpecs
+    mergeKernelSpecs: (kernelSpecs) ->
+        _.assign @_kernelSpecs, kernelSpecs
 
-        Config.setJson 'kernelspec', kernelspecs: kernelSpecs
 
-        message = 'Hydrogen Kernels updated:'
-        options = detail: (_.map kernelSpecs, 'spec.display_name').join('\n')
-        atom.notifications.addInfo message, options
-
-
-    updateKernelSpecs: ->
-        commands = [
-            'jupyter kernelspec list --json --log-level=CRITICAL',
-            'ipython kernelspec list --json --log-level=CRITICAL',
-        ]
-
-        child_process.exec commands[0], (err, stdout, stderr) =>
+    updateKernelSpecs: (callback) ->
+        @_kernelSpecs = @getKernelSpecsFromSettings
+        @getKernelSpecsFromJupyter (err, kernelSpecsFromJupyter) =>
             unless err
-                @saveKernelSpecs stdout
+                @mergeKernelSpecs kernelSpecsFromJupyter
+
+            if _.isEmpty @_kernelSpecs
+                message = 'No kernel specs found'
+                options =
+                    detail: 'Use kernelSpec option in Hydrogen or update
+                    IPython/Jupyter to a version that supports: `jupyter
+                    kernelspec list --json` or `ipython kernelspec list --json`'
+                    dismissable: true
+                atom.notifications.addError message, options
+            else
```

Comment:
I would add `err = null` here so that an error will be returned only if `@_kernelSpecs` is empty